docs/manual-EN/1.overview/3.design-and-architecture/2.storage-design.md

```diff
@@ -74,69 +73,77 @@ Pict.5 In-key format
+When transferring leadership, it is worth noting the timing when a leader abandons the leadership and when all the followers start a leader election. When a `transfer leadership` command is committed, from the leader's view, it loses the leadership. From other followers's view, when receiving this command, it starts a new leader election. These two operations must be executed in the same process with a normal raft leader election. Otherwise, some corner cases can occur and they are very hard to test.
 
 ### Membership Change
 
-To avoid split-brain, when members in a Raft Group change, an intermediate state is required. In such state, the majority of the old group and  new group always have an overlap, thus preventing the old or new group from making decisions unilaterally. This is the `joint consensus` mentioned in the thesis. To make it even simpler, Diego Ongaro suggests **adding or removing a peer once to ensure the overlap between the majority of the new group and the old group** in his doctoral thesis. **Nebula Graph**'s implementation also uses this approach, except that the way to add or remove member is different. For details, please refer to `addPeer/removePeer` in Raft Part class.
+To avoid the brain-split, when Raft Group members changed, an intermediate state is required. In such state, the majority of the old group and new group always have an overlap. This majority overlap will prevent neither group from making decisions unilaterally. This is the `joint consensus` as mentioned in the famous Raft thesis. To make it even simpler, Diego Ongaro suggests to **add or remove only one peer at a time to ensure the overlap between the majority** in his doctoral thesis. **Nebula Graph**'s implementation also uses this approach, except that the implementation to add or remove member is different. For details, please refer to `addPeer/removePeer` in Raft Part class.
 
 ### Snapshot
 
-When the leader changes during sending snapshot, the follower may only receive half of the snapshot data, so partition data cleanup is required. Because multiple partitions share a single storage, how to clean up the data is a cumbersome problem. In addition, during the snapshot process, a large amount of IO is generated. Considering the performance, we do not want this process to share an IO threadPool with the normal Raft, and the process requires a large amount of memory, thus how to optimize the memory usage is critical for performance.
+Take snapshot is a common command during daily DBA operations. But snapshot operation will introduce extra challenges when consider together with the raft protocol. It's very error prone.
```

Comment:
Snapshot is a common command during daily DBA operations. But snapshot operation will introduce extra challenges when considering together with the raft protocol. It's very error-prone.